src/test/java/uk/gov/hmcts/reform/notificationservice/service/NotificationServiceTest.java

```diff
@@ -102,6 +105,69 @@ void should_leave_notification_record_as_is_when_relevant_exception_from_client_
         verify(notificationRepository, never()).markAsFailure(notification.id);
     }
 
+    @Test
+    void should_return_notifications_for_file_name_and_service() {
+        // given
+        final String zipFileName = "zip_file_name";
+        final String service = "service";
+
+        var notification1 = new Notification(
+            1L,
+            "notificationId1",
+            zipFileName,
+            "po_box1",
+            service,
+            "DCN1",
+            ErrorCode.ERR_METAFILE_INVALID,
+            "invalid metafile1",
+            Instant.now(),
+            Instant.now(),
+            NotificationStatus.SENT
+        );
+        var notification2 = new Notification(
+            2L,
+            "notificationId2",
+            zipFileName,
+            "po_box2",
+            service,
+            "DCN2",
+            ErrorCode.ERR_FILE_LIMIT_EXCEEDED,
+            "invalid metafile2",
+            Instant.now(),
+            Instant.now(),
+            NotificationStatus.SENT
+        );
+        given(notificationRepository.find(zipFileName, service))
+                  .willReturn(asList(notification1, notification2));
+
+        // when
+        var notificationResponses = notificationService.findByFileNameAndService(zipFileName, service);
+
+        // then
+        assertThat(notificationResponses)
+            .hasSize(2)
+            .usingFieldByFieldElementComparator()
+            .containsExactlyInAnyOrder(
+                toNotificationResponse(notification1),
```

Comment:
Mapping method in service is verified with mapping method in test `toNotificationResponse`.
Maybe it would be better to verify field by field.

Reply:
Have done using tuple